tests/loki/loki_directbeam_test.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from typing import Callable, List
+
+import numpy as np
+import sciline
+import scipp as sc
+from loki_common import make_params
+
+import esssans as sans
+from esssans.types import FinalDims, QBins
+
+
+def get_I0(q_loc: sc.Variable) -> sc.Variable:
+    from esssans.loki.data import get_path
+
+    data = np.loadtxt(get_path('PolyGauss_I0-50_Rg-60.txt'))
+    qcoord = sc.array(dims=["Q"], values=data[:, 0], unit='1/angstrom')
+    theory = sc.DataArray(
+        data=sc.array(dims=["Q"], values=data[:, 1], unit=''), coords={"Q": qcoord}
+    )
+    ind = np.argmax((qcoord > q_loc).values)
+    I0 = (theory.data[ind] - theory.data[ind - 1]) / (qcoord[ind] - qcoord[ind - 1]) * (
+        q_loc - qcoord[ind - 1]
+    ) + theory.data[ind - 1]
+    return I0
```

Comment:
Same (or very similar) code is found in the docs notebook. Should this become part of the package?

Reply:
Should I build a small pipeline that is independent of the $I(Q)$ pipeline which would compute the direct beam function? It would have a provider for I0, and the filename as a parameter?

The provider for the `DirectBeamFunction` would have the iterations loop inside it, instead of just calling a pure python function as it is now.

Reply:
Not everything in the package has to use Sciline. It could just be a function (but it could also be written like a provider, if we want to make a pipeline later).